storagemanager/users/models.py

```diff
@@ -1,16 +1,57 @@
 from django.db import models
-from django.contrib.auth.models import User
+from django.contrib.auth.models import AbstractUser, AbstractBaseUser, BaseUserManager
 from PIL import Image
+from django.conf import settings
 
 
-class Profile(models.Model):
-    user = models.OneToOneField(User, on_delete=models.CASCADE)
-    phone_number = models.CharField(max_length=15)
+class UserManager(BaseUserManager):
+    def create_user(self, email, password=None):
+        """
+        Creates and saves an user with the given email
+        and password.
+        """
+        if not email:
+            raise ValueError('Users must have an email address')
+        if not password:
+            raise ValueError('Users must have a password')
+
+        user = self.model(
+            email=self.normalize_email(email)
+        )
+        user.set_password(password)
+        user.save(using=self._db)
+        return user
+
+    def create_superuser(self, email, password=None):
+        """
+        Creates and saves a superuser with the given email
+        and password.
+        """
+        user = self.create_user(email, password=password)
+        user.admin = True
+        user.staff = True
+        user.save(using=self._db)
+        return user
+
+
+class User(AbstractBaseUser):
+    email = models.EmailField(max_length=255, unique=True)
+    full_name = models.CharField(
+        max_length=255, blank=True, null=True)
+    phone_number = models.CharField(max_length=15, blank=True, null=True)
     image = models.ImageField(
         default='default_profile_pic.jpg', upload_to='profile_pics')
+    active = models.BooleanField(default=True)
+    staff = models.BooleanField(default=False)
+    admin = models.BooleanField(default=False)
+
+    objects = UserManager()
+
+    USERNAME_FIELD = 'email'
+    REQUIRED_FIELDS = []
```

Comment:
What these two properties do?

Reply:
USERNAME_FIELD contains name of the field, which will be used as unique identifier for every user in db.

REQUIRED_FIELDS is a list containing names of the fields, which are gonna be prompt while creating superuser (beside of field from USERNAME_FIELD and password). I removed it in newest commit, since it was an empty list.

Reply:
You haven't used `USERNAME_FIELD` anywhere so I'm not sure what are your intentions with it but thinking if it is right place for it. Why can't you just refer to `User.email`? Or have it outside of model itself as a i.e. `USER_UNIQUE_FIELD_NAME`?